app/src/main/java/com/babylon/wallet/android/presentation/transaction/TransactionReviewViewModel.kt

```diff
@@ -252,6 +256,7 @@ class TransactionReviewViewModel @Inject constructor(
         private val latestFeesMode: Sheet.CustomizeFees.FeesMode = Sheet.CustomizeFees.FeesMode.Default,
         val error: UiMessage? = null,
         val isNoMnemonicErrorVisible: Boolean = false,
+        val isDepositRulesErrorVisible: Boolean = false,
```

Comment:
This will soon get out of hand, if we keep it like this. Maybe a different structure would be more suited. Or not needed at all. 

What if we kept the same structure of `error: UiMessage` and on UI check which kinds of error we should display as dialogs and which we should display as snackbars. It is a UI thing after all.

So we could remove from state both `isNoMnemonicErrorVisible` and `isDepositRulesErrorVisible` and still keep `error: UiMessage`.

On UI we could 

```
if (error.isPreviewedInDialog) {
    Dialog(....)
} else {
    SnackbarUIMessage(
        message = state.error,
        snackbarHostState = snackBarHostState,
        onMessageShown = onMessageShown
    )
}
```

And in State we could:
data class State(...) {
    val UiMessage.isPreviewedInDialog: Boolean
            get() = error....
}

Reply:
> This will soon get out of hand, if we keep it like this. Maybe a different structure would be more suited. Or not needed at all.
> 
> What if we kept the same structure of `error: UiMessage` and on UI check which kinds of error we should display as dialogs and which we should display as snackbars. It is a UI thing after all.
> 
> So we could remove from state both `isNoMnemonicErrorVisible` and `isDepositRulesErrorVisible` and still keep `error: UiMessage`.
> 
> On UI we could
> 
> ```
> if (error.isPreviewedInDialog) {
>     Dialog(....)
> } else {
>     SnackbarUIMessage(
>         message = state.error,
>         snackbarHostState = snackBarHostState,
>         onMessageShown = onMessageShown
>     )
> }
> ```
> 
> And in State we could: data class State(...) { val UiMessage.isPreviewedInDialog: Boolean get() = error.... }

I see, makes sense, let me make it a bit better 👍

Reply:
@micbakos-rdx Pushed the fix. Let me know if its makes sense or if it can be improved.